modules/base/src/algorithm/camerautils.cpp

```diff
@@ -0,0 +1,210 @@
+        case Side::XPositive:
+            return vec3(0, 1, 0);
+        case Side::YNegative:
+            return vec3(1, 0, 0);
+        case Side::YPositive:
+            return vec3(1, 0, 0);
+        case Side::ZNegative:
+            return vec3(0, 1, 0);
+        case Side::ZPositive:
+            return vec3(0, 1, 0);
+        default:
+            return vec3(0;
+    }
+}
+
+}  // namespace detail
+
+void setCameraView(CameraProperty &cam, const mat4 &dataToWorld, Side side, float scale,
+                   bool setNearFar, bool setLookRanges) {
+    if (auto perspectiveCamera = dynamic_cast<PerspectiveCamera *>(&cam.get())) {
+
+        auto viewDir = detail::getViewDir(side);
+        auto lookUp = detail::getLookUp(side);
+
+        const float offset = .5f;
+        auto lookTo = vec3(dataToWorld * vec4(vec3(offset), 1.f));
+        auto sideDir = glm::cross(viewDir, lookUp);
+
+        auto frontPoint = vec3(dataToWorld * vec4(vec3(offset) - (viewDir * offset), 1.f));
+        auto sidePoint =
+            vec3(dataToWorld * vec4(vec3(offset) + (sideDir * offset) - (viewDir * offset), 1.f));
+        auto topPoint =
+            vec3(dataToWorld * vec4(vec3(offset) + (lookUp * offset) - (viewDir * offset), 1.f));
+
+        float fovy = perspectiveCamera->getFovy() / 2.0f;
+        float fovx = glm::degrees(std::atan(cam.getAspectRatio() * std::tan(glm::radians(fovy))));
+
+        float h1 = glm::distance(frontPoint, topPoint) * scale;
+        float d1 = h1 * std::tan(glm::radians(90 - fovy));
+
+        float h2 = glm::distance(frontPoint, sidePoint) * scale;
+        float d2 = h2 * std::tan(glm::radians(90 - fovx));
+
+        auto lookFrom = frontPoint + std::max(d1, d2) * glm::normalize(frontPoint - lookTo);
+        auto camUp = glm::normalize(topPoint - frontPoint);
+
+        NetworkLock lock(&cam);
+
+        if (setNearFar) {
+            setCameraNearFar(cam, dataToWorld);
+        }
+        if (setLookRanges) {
+            setCameraLookRanges(cam, dataToWorld);
+        }
+
+        cam.setLook(lookFrom, lookTo, camUp);
+    } else {
+        LogWarnCustom("camerautil::setCameraView",
+                      "setCameraView only supports perspective cameras");
+    }
+}
+
+void setCameraLookRanges(CameraProperty &cam, const mat4 &dataToWorld, float zoomRange) {
+    auto m = dataToWorld;
+
+    vec3 lookTo(dataToWorld * vec4(vec3(.5f), 1.f));
+
+    vec3 dx(dataToWorld[0]);
+    vec3 dy(dataToWorld[1]);
+    vec3 dz(dataToWorld[2]);
+
+    auto p0 = lookTo + (dx + dy + dz) * zoomRange;
+    auto p1 = lookTo - (dx + dy + dz) * zoomRange;
+    cam.lookFrom_.setMinValue(glm::min(p0, p1));
+    cam.lookFrom_.setMaxValue(glm::max(p0, p1));
+    p0 = lookTo + (dx + dy + dz);
+    p1 = lookTo - (dx + dy + dz);
+    cam.lookTo_.setMinValue(glm::min(p0, p1));
+    cam.lookTo_.setMaxValue(glm::max(p0, p1));
+}
+
+std::pair<float, float> computeCameraNearFar(const mat4 &dataToWorld, float zoomRange,
```

Comment:
How does this compare to meshutil:computeNearFarPlanes?
We should probably only have one of them

Reply:
I found that one after implementing this one. That one now calls this one.

Reply:
The new code is not equivalent, it does not consider offset. Have you tested this with object that are not centered around the world origin?

Reply:
near and far plane does not need to consider offset, they are in view space

Reply:
yes, I have tested this with many volumes, both centered and not centered, with regular basis and rotational basis.

Reply:
Hmm, that is weird. If an object is far away from the camera it is necessary to have a large far distance so it is a combination of camera and object placement. I have some nifty volumes with large offsets that we might be able to try it on and where it failed before I made the other implementation